src/processor/operator/persistent/reader/rdf/rdf_reader.cpp

```diff
@@ -229,27 +178,59 @@ SerdStatus RdfReader::rrlHandle(void* handle, SerdStatementFlags, const SerdNode
     return SERD_SUCCESS;
 }
 
+static void addNode(std::vector<std::string>& vector, const SerdNode* node) {
+    vector.emplace_back((const char*)node->buf, node->n_bytes);
+}
+
+SerdStatus RdfReader::allHandle(void* handle, SerdStatementFlags, const SerdNode*,
+    const SerdNode* subject, const SerdNode* predicate, const SerdNode* object,
+    const SerdNode* object_datatype, const SerdNode*) {
+    if (!supportTriple(subject, predicate, object)) {
+        return SERD_SUCCESS;
+    }
+    auto reader = reinterpret_cast<RdfReader*>(handle);
+    if (object->type == SERD_LITERAL) {
+        addNode(reader->store->literalTripleStore.subjects, subject);
+        addNode(reader->store->literalTripleStore.predicates, predicate);
+        addNode(reader->store->literalTripleStore.objects, object);
+        if (object_datatype != nullptr) {
+            addNode(reader->store->literalTripleStore.objectType, object_datatype);
+        } else {
+            reader->store->literalTripleStore.objectType.push_back("");
+        }
+    } else {
+        addNode(reader->store->resourceTripleStore.subjects, subject);
+        addNode(reader->store->resourceTripleStore.predicates, predicate);
+        addNode(reader->store->resourceTripleStore.objects, object);
+    }
+    return SERD_SUCCESS;
+}
+
 SerdStatus RdfReader::prefixHandle(void* handle, const SerdNode* /*name*/, const SerdNode* uri) {
     auto reader = reinterpret_cast<RdfReader*>(handle);
     reader->currentPrefix = reinterpret_cast<const char*>(uri->buf);
     return SERD_SUCCESS;
 }
 
-offset_t RdfReader::read(DataChunk* dataChunk) {
+offset_t RdfReader::read(DataChunk* dataChunk, RdfStore* store_) {
     if (status) {
         return 0;
     }
 
+    store = store_;
     switch (mode) {
     case RdfReaderMode::RESOURCE_TRIPLE:
     case RdfReaderMode::LITERAL_TRIPLE: {
         sVector = dataChunk->getValueVector(0).get();
         pVector = dataChunk->getValueVector(1).get();
         oVector = dataChunk->getValueVector(2).get();
     } break;
-    default: {
+    case RdfReaderMode::RESOURCE:
+    case RdfReaderMode::LITERAL: {
         sVector = dataChunk->getValueVector(0).get();
     } break;
+    default:
+        break;
     }
     vectorSize = 0;
```

Comment:
Double check if we really need `vectorSize`. Updating selectedSize might be doable inside reader.